File: 2022/2022_03_cpp/220326.cpp
//
// Created by z on 3/26/2022.
// # 704

class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size() - 1, ret = -1;
        while (l <= r)
        {
            int m = (l + r) / 2;
            if (target == nums[m])
                return m;
            else if (target > nums[m])
                l = m+1;
            else
                r = m-1;
        }
        return ret;
    }
};

File: 2022/2022_python/220326.py
#
# created by z on 3/26/2022
# 704

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l: int = 0
        r: int = len(nums) - 1
        while l <= r:
            m: int = int((l + r) / 2)
            if nums[m] == target:
                return m
            elif nums[m] > target:
                r = m - 1
            else:
                l = m + 1

        return -1
